docs/Contributors_Guide/continuous_integration.rst

```diff
@@ -1,115 +1,376 @@
+was saved as a GitHub Secret.
+
+.. _cg-ci-testing-workflow:
+
+Testing Workflow
+================
+
 Name
-^^^^
+----
 
 The name of a workflow can be specified to describe an overview of what is run.
-Currently METplus only has 1 workflow, but others can be added. The following
-line in the testing.yml file::
+The following line in the testing.yml file::
 
-    name: METplus CI/CD Workflow
+    name: Testing
 
-defines the workflow that runs all of the jobs.
+defines the workflow identifier that can be seen from the Actions tab on the
+METplus GitHub page.
 
 .. figure:: figure/gha-workflow-name.png
 
 Event Control
-^^^^^^^^^^^^^
+-------------
+
+The **on** keyword is used to determine which events will trigger the workflow
+to run. There are currently 3 types of events that trigger this workflow:
+push, pull_request, and workflow_dispatch.
+The jobs that are run in this workflow depend on which event has triggered it.
+There are a lot of jobs that are common to multiple events.
+To avoid creating multiple workflow .yml files that contain redundant jobs,
+an additional layer of control is added within this workflow.
+See :ref:`cg-ci-job-control` for more information.
+
+Push
+^^^^
 
-The "on" keyword is used to determine which events will trigger the workflow
-to run::
+::
 
     on:
+
       push:
         branches:
           - develop
           - develop-ref
-          - feature_*
-          - main_*
-          - bugfix_*
+          - 'feature_*'
+          - 'main_*'
+          - 'bugfix_*'
+        paths-ignore:
+          - 'docs/**'
+
+This configuration tells GitHub Actions to trigger the workflow when changes
+are pushed to the repository and the following criteria are met:
+
+* The branch is named **develop** or **develop-ref**
+* The branch starts with **feature\_**, **main\_**, or **bugfix\_**
+* Changes were made to at least one file that is not in the **docs** directory.
+
+Pull Request
+^^^^^^^^^^^^
+
+::
+
       pull_request:
         types: [opened, reopened, synchronize]
+        paths-ignore:
+          - 'docs/**'
+
+This configuration tells GitHub Actions to trigger the workflow for
+pull requests in the repository and the following criteria are met:
 
-This configuration tells GitHub Actions to trigger the workflow when:
+* The pull request was opened, reopened, or synchronized.
+* Changes were made to at least one file that is not in the **docs** directory.
 
-* A push event occurs on the develop or develop-ref branch
-* A push event occurs on a branch that starts with
-  feature\_, main\_, or bugfix\_
-* A pull request is opened, reopened, or synchronized (when new changes are
-  pushed to the source branch of the pull request.
+The **synchronize** type triggers a workflow for every push to a branch
+that is included in an open pull request.
+If changes were requested in the pull request review,
+a new workflow will be triggered for each push.
+To prevent many workflows from being triggered, the pull request
```

Comment:
Consider editing with:
```
To prevent many workflows from being triggered, developers are encouraged to limit the number of pushes for open pull requests. Note that pull requests can be closed until the necessary changes are completed or [Commit Message Keywords](https://metplus.readthedocs.io/en/feature_675_ci_doc/Contributors_Guide/continuous_integration.html#cg-ci-commit-message-keywords) can be used to suppress the testing workflow.
```